stage_tester/locustfile.py

```diff
@@ -0,0 +1,122 @@
+        self.client_id = cfg['load_test']['client_id']
+        self.api_url = cfg['load_test']['api_url']
+        self.login_url = cfg['load_test']['login_url']
+        self.organization = cfg['load_test']['organization']
+        
+        self.clusters = []
+                    
+
+    def login(self):
+        
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+        }
+        payload = "grant_type=refresh_token&client_id={}&refresh_token={}".format(
+            self.client_id,
+            self.offline_token
+        )
+        response = requests.post(self.login_url, headers=headers, data=payload)
+        self.access_token = response.json()['access_token']
+
+
+    @task
+    def retrieve_org_overview(self):
+
+        self.get_api_call(
+            '/api/insights-results-aggregator/v1/org_overview', self.access_token)
+
+
+    @task
+    def retrieve_gathering_conditions(self):
+
+        self.get_api_call(
+            '/api/gathering/gathering_rules', self.access_token)
+
+
+    @task
+    def retrieve_cluster_list(self):
+        
+        path = f'/api/insights-results-aggregator/v1/organizations/{self.organization}/clusters'
+        response = self.get_api_call(path, self.access_token)
+        payload = response.json()
+        if "clusters" in payload:
+            self.clusters = payload["clusters"]
+
+    @task
+    def retrieve_additional_info(self):
+        self.get_api_call(
+            '/api/insights-results-aggregator/v1/info', self.access_token)
+
+    @task
+    def retrieve_results_for_cluster(self):
+        
+        if self.clusters:
+            uuid = random.choice(self.clusters)
+            self.get_api_call(
+                f'/api/insights-results-aggregator/v1/clusters/{uuid}/report', self.access_token)
+
+
+    def get_api_call(self, path, token):
+        
+        headers = {
+            'Accept': 'application/json',
+            'Authorization': f"Bearer {token}",
+        }
+        response = self.client.get(path, headers=headers)
+
+        # elementary check for response content
+        assert response is not None, "Proper response expected"
+
+        return response
+
+
+@events.quitting.add_listener
+def _(environment, **kw):
+    if environment.stats.total.fail_ratio > 0.1:
+        logging.error("Test failed due to failure ratio > 10%")
+        environment.process_exit_code = 1
+    elif environment.stats.total.avg_response_time > 500:
+        logging.error("Test failed due to average response time ratio > 500 ms")
+        environment.process_exit_code = 1
+    elif environment.stats.total.get_response_time_percentile(0.95) > 1000:
+        logging.error("Test failed due to 95th percentile response time > 1 s")
+        environment.process_exit_code = 1
+    else:
+        environment.process_exit_code = 0
+
+if __name__ == '__main__':
+
+    print("start load testing")
+    
+    run_single_user(ApiUser)
```

Comment:
According to the docs, `run_single_user` is for debugging purposes. I know we only have one test user so far, but just wanted to raise this in case you weren't aware.

Reply:
yes, in fact our test will use 10 users (but same account)

now that i think of it we could parametrize the users number...